addons/hr_holidays/models/hr_leave.py

```diff
@@ -539,7 +539,7 @@ def _get_durations(self, check_leave_type=True, resource_calendar=None):
         for leave in employee_leaves:
             if not leave.date_from or not leave.date_to:
                 continue
-            employees_by_dates_calendar[(leave.date_from, leave.date_to, resource_calendar or leave.resource_calendar_id)] += leave.employee_id
+            employees_by_dates_calendar[(leave.date_from, leave.date_to, leave.holiday_status_id, resource_calendar or leave.resource_calendar_id)] += leave.employee_id
```

Comment:
I think it would be better to only include the value of the field (the boolean value), such that different leave types with the same value for this field, still get batched together.